swim/heal_partition.go

```diff
@@ -72,19 +72,35 @@ func nodesThatNeedToReincarnate(MA, MB []Change) (changesForA, changesForB []Cha
 		// If a node would be overwritten and would stop being pingable, it is
 		// not suited for merging.
 		if b.isPingable() && a.overrides(b) && !a.isPingable() {
-			changesForB = append(changesForB, Change{
-				Address:     a.Address,
-				Incarnation: a.Incarnation,
-				Status:      Suspect,
-			})
+			// take a look at the state of the member
+			member := Member{}
+			member.populateFromChange(&a)
+
+			// create the member into a change
+			change := Change{}
+			change.populateSubject(&member)
```

Comment:
It might be nicer/more readable to just copy the existing change (`a`) and change the status to suspect or have a `change.populateFromChange`-method.